base/loading.jl

```diff
@@ -557,10 +557,14 @@ function pkgversion(m::Module)
     path = pkgdir(m)
     path === nothing && return nothing
     @lock require_lock begin
-        v = get_pkgversion_from_path(path)
         pkgorigin = get(pkgorigins, PkgId(moduleroot(m)), nothing)
         # Cache the version
-        if pkgorigin !== nothing && pkgorigin.version === nothing
+        if pkgorigin !== nothing && pkgorigin.version !== nothing
+            return pkgorigin.version
+        end
+        v = get_pkgversion_from_path(path)
+        # Cache the version
+        if pkgorigin !== nothing
```

Comment:
Could `pkgdir` return `nothing` for sysimage modules that might have `pkgorigin.version` populated? Should we move that check into the lock?